### PR TITLE
feat: await sendPaymentResult transaction expiration check

### PR DESCRIPTION
**Pull request by pietro-tota**
Ecommerce local PR for update environment --> https://github.com/pagopa/pagopa-ecommerce-local/pull/75

#### List of Changes
Add logic to skip expiration for those transactions stuck in CLOSED status for which an OK response has been received and that are still waiting for a send payment result according to the `PENDING_TRANSACTIONS_SEND_PAYMENT_RESULT_TIMEOUT_SECONDS` timeout parameter

#### Motivation and Context
Those modifications are needed in order to handle transactions that are stuck in CLOSED status waiting for send payment result

#### How Has This Been Tested?
Junit tests + local tests using ecommerce local

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by giovanniberti — APPROVED** [review]

**Comment by giovanniberti** [comment]
Shouldn't this include a `sendPaymentResult` offset like in https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/99

File: src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/transactionanalyzer/PendingTransactionAnalyzer.kt [hunk 1]
```diff
@@ -82,22 +91,45 @@ class PendingTransactionAnalyzer(
...
-                val sendExpiryEvent = transactionStatusesForSendExpiryEvent.contains(it.status)
-                logger.info(
-                    "Transaction with id: [${it.transactionId}] state: [${it.status}], expired transaction statuses: ${transactionStatusesForSendExpiryEvent}. Send event: $sendExpiryEvent"
-                )
-                if (sendExpiryEvent) {
-                    Mono.just(it)
-                } else {
-                    Mono.empty()
+            .filterWhen { baseTransaction ->
+                val skipEvent =
```
[the remark this answers is not in the document]
Comment (pietro-tota) [reply src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/transactionanalyzer/PendingTransactionAnalyzer.kt]: Fixed with https://github.com/pagopa/pagopa-ecommerce-transactions-scheduler-service/pull/32/commits/641ecff28442978ea346169302a1d8de0d62f7b0

**Comment by pietro-tota** [comment]
> Shouldn't this include a `sendPaymentResult` offset like in [pagopa/pagopa-ecommerce-event-dispatcher-service#99](https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/99)?

Offset in the above PR was add in order to avoid sending multiple expiration events also for transactions that are in a state for which only few seconds are left for send payment result to be received.

I don't think it can be useful here since transactions are analyzed twice by batch (because of the sliding window that take into account also the previuos batch execution) so if a transaction is still waiting for send payment result in execution Tx then will not in execution Tx+1, so it will be covered by the next batch execution. 

With that in mind I think that maybe it can be useless to add here offset.

What do you think?

P.S. in case we will change send payment result outcome timeout we have to take into account also batch executions time window multiplier so that the same transaction is covered, at least, by two executions.

**Comment by giovanniberti** [comment]
> With that in mind I think that maybe it can be useless to add here offset.
Agree :)

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**